rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/SchedulerClient.java

```diff
@@ -987,27 +988,29 @@ public boolean deleteFile(String space, String pathname) throws NotConnectedExce
 
     @Override
     public void renewSession() throws NotConnectedException {
-        Closer closer = Closer.create();
-        try {
-            LoginForm loginForm = new LoginForm();
-            loginForm.setUsername(connectionInfo.getLogin());
-            loginForm.setPassword(connectionInfo.getPassword());
-            if (connectionInfo.getCredentialFile() != null) {
-                FileInputStream inputStream = new FileInputStream(connectionInfo.getCredentialFile());
+
+        LoginForm loginForm = new LoginForm();
+        loginForm.setUsername(connectionInfo.getLogin());
+        loginForm.setPassword(connectionInfo.getPassword());
+
+        if (connectionInfo.getCredentialFile() != null) {
+            try (Closer closer = Closer.create();
+                    FileInputStream inputStream = new FileInputStream(connectionInfo.getCredentialFile())) {
+
                 closer.register(inputStream);
                 loginForm.setCredential(inputStream);
-            }
-            sid = restApi().loginOrRenewSession(sid, loginForm);
 
-        } catch (Exception e) {
-            throw new RuntimeException(e);
-        } finally {
-            try {
-                closer.close();
             } catch (IOException e) {
-                // ignore
+                throw new RuntimeException(e);
             }
         }
+
+        try {
+            sid = restApi().loginOrRenewSession(sid, loginForm);
+        } catch (KeyException | SchedulerRestException | LoginException | NotConnectedRestException e) {
+            throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)